geonode/base/models.py

```diff
@@ -777,7 +777,7 @@ class ResourceBase(PolymorphicModel, PermissionLevelMixin, ItemBase):
     extra_metadata_help_text = _(
         'Additional metadata, must be in format [ {"metadata_key": "metadata_value"}, {"metadata_key": "metadata_value"} ]')
     # internal fields
-    uuid = models.CharField(max_length=36, unique=True, default=str(uuid.uuid4))
+    uuid = models.CharField(max_length=36, unique=True, default=uuid.uuid4)
```

Comment:
@afabiani this fix is needed since the last migration available for the UUID, is marked as a string of the function itself and not a result.
https://github.com/GeoNode/geonode/blob/a41fbfdf9597b4a33f20fa242b5893b678fbbccb/geonode/base/migrations/0080_alter_resourcebase_uuid.py#L17

This end-up up having an error because the system tries to save the UUID as a "function of...." 

This fix, will let geonode decide the UUID (as object) and the fix in the save_method below, helps it to convert it to string if needed